### DISCUSSION TITLE
MQTT unsubscribe is, AFAIK, not yet implemented. What's the best alternative?

### DISCUSSION THREAD

**Discussion by lazyoldjack** [toitlang/toit#750] [Q&A] [answered]
I wish to pick up a retained value using an MQTT subscription to a particular topic when my ESP32 is first powered up. I then wish to **NOT** receive anything from that topic until the next power up because this is the only time it's relevant.

Is my only option to carry on receiving that topic and just throw away the data? That's fine, but seems wasteful of resources on both the ESP32 and my MQTT Broker.

**Comment by kasperl** [comment]
I think the easiest is that we just go ahead and implement the unsubscribe functionality. Thoughts, @floitsch?

**Comment by lazyoldjack** [comment]
Well, obviously that gets my vote :)

Easy for me so say though...

------------------------------------------------------------------------


On 21/05/2022 08:48, Kasper Lund wrote:
>
> I think the easiest is that we just go ahead and implement the 
> unsubscribe functionality. Thoughts, @floitsch 
> <https://github.com/floitsch>?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/toitlang/toit/discussions/750#discussioncomment-2795034>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AFADTMTRLACULGKUVR5MOL3VLCIOBANCNFSM5WROGNTQ>.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>

-- 
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

**Comment by floitsch** [comment, marked as answer]
I will implement it today.

Edit: unsubscribe is now implemented and committed.

**Reply by floitsch** [reply]
https://github.com/toitware/mqtt/pull/20

**Reply by lazyoldjack** [reply]
Brilliant, thank you. Working and I'm using it!
